packages/app/src/visualizer/Visualizer.tsx

```diff
@@ -1,37 +1,37 @@
 import type { Entity } from '@h5web/shared';
+import { assertDefined } from '@h5web/shared';
 
 import Profiler from '../Profiler';
+import { useActiveVis } from '../vis-packs/hooks';
 import type { VisDef } from '../vis-packs/models';
 import VisSelector from './VisSelector';
 import styles from './Visualizer.module.css';
 
-interface Props<T extends VisDef> {
+interface Props {
   entity: Entity;
-  activeVis: T | undefined;
-  supportedVis: T[];
-  onActiveVisChange: (vis: T) => void;
+  supportedVis: VisDef[];
 }
 
-function Visualizer<T extends VisDef>(props: Props<T>) {
-  const { entity, activeVis, supportedVis, onActiveVisChange } = props;
+function Visualizer(props: Props) {
+  const { entity, supportedVis } = props;
+  assertDefined(supportedVis[0], 'Expected supported visualization');
 
-  if (!activeVis) {
-    return (
-      <p className={styles.fallback}>
-        No visualization available for this entity.
-      </p>
+  const [activeVis, setActiveVis] = useActiveVis(supportedVis);
```

Comment:
`useActiveVis` is now called here instead of duplicated in the now-removed `CorePack` and `NexusPack`. It is now guaranteed to return a visualization.